odxtools/diagcomm.py

```diff
@@ -216,12 +216,12 @@ def _resolve_snrefs(self, context: SnRefContext) -> None:
         if TYPE_CHECKING:
             diag_layer = odxrequire(context.diag_layer)
             self._protocols = NamedItemList([
-                resolve_snref(prot_snref, diag_layer.protocols, DiagLayer)
+                resolve_snref(prot_snref, getattr(diag_layer, "protocols", []), Protocol)
                 for prot_snref in self.protocol_snrefs
             ])
         else:
             diag_layer = odxrequire(context.diag_layer)
             self._protocols = NamedItemList([
-                resolve_snref(prot_snref, diag_layer.protocols)
+                resolve_snref(prot_snref, getattr(diag_layer, "protocols", []))
```

Comment:
Avoid `getattr`, it allows for typing bugs that mypy can't catch, use isinstance instead

Reply:
I just tried this: This is pretty difficult to do as part of a list comprehension and it would not come with a major benefit because `resolve_snref(..., Protocol)`  returns a protocol object from the type checking perspective.